lib/danger/request_sources/github.rb

```diff
@@ -108,13 +108,16 @@ def generate_github_description(warnings: nil, errors: nil)
     def generate_comment(warnings: nil, errors: nil, messages: nil)
       require 'erb'
 
-      @warnings = warnings
-      @errors = errors
-      @messages = messages
-
       md_template = File.join(Danger.gem_path, "lib/danger/comment_generators/github.md.erb")
-      comment = ERB.new(File.read(md_template)).result(binding) # http://www.rrn.dk/rubys-erb-templating-system
-      return comment
+
+      # erb: http://www.rrn.dk/rubys-erb-templating-system
+      # for the extra args: http://stackoverflow.com/questions/4632879/erb-template-removing-the-trailing-line
+      @tables = [
+        { :name => "Error", :emoji => "no_entry_sign", :content => errors },
+        { :name => "Warning", :emoji => "warning", :content => warnings },
+        { :name => "Message", :emoji => "book", :content => messages }
+      ]
+      return ERB.new(File.read(md_template), 0, "-").result(binding)
```

Comment:
>      # for the extra args: http://stackoverflow.com/questions/4632879/erb-template-removing-the-trailing-line

roughly, if you do `-%>` now, then it will _not_ add a newline straight after it - which was something you were struggling with